contracts/NodeVersionBeacon.cdc

```diff
@@ -116,94 +120,133 @@ pub contract ExecutionNodeVersionBeacon {
+            //emit NodeVersionTableAddition(height: targetBlockHeight, version: newVersion)
         }
 
         /// Deletes the last entry in versionTable which defines an upcoming version boundary
         /// Could be helpful in case rollback is needed
         pub fun deleteUpcomingVersionBoundary(blockHeight: UInt64): Semver {
             pre {
-                ExecutionNodeVersionBeacon.versionTable.keys.contains(blockHeight): "No boundary defined at that blockHeight."
-                ExecutionNodeVersionBeacon.upcomingBlockBoundaries.contains(blockHeight): "Boundary not defined in upcomingBlockBoundaries."
+                NodeVersionBeacon.versionTable.keys.contains(blockHeight): "No boundary defined at that blockHeight."
+                NodeVersionBeacon.upcomingBlockBoundaries.contains(blockHeight): "Boundary not defined in upcomingBlockBoundaries."
             }
 
             // Ensure deletion occurs with enough time for nodes to respond
             assert(
-                blockHeight > getCurrentBlock().height + ExecutionNodeVersionBeacon.versionUpdateBuffer,
+                blockHeight > getCurrentBlock().height + NodeVersionBeacon.versionUpdateBuffer,
                 message: "Cannot delete version boundary within update buffer period or historical mappings."
             )
 
             // Remove the version mapping and upcomingBlockBoundaries then emit event
-            let removed: Semver = ExecutionNodeVersionBeacon.versionTable.remove(
+            let removed: Semver = NodeVersionBeacon.versionTable.remove(
                 key: blockHeight
             )!
-            ExecutionNodeVersionBeacon.upcomingBlockBoundaries.remove(
-                at: ExecutionNodeVersionBeacon.upcomingBlockBoundaries.firstIndex(of: blockHeight)!
+            NodeVersionBeacon.upcomingBlockBoundaries.remove(
+                at: NodeVersionBeacon.upcomingBlockBoundaries.firstIndex(of: blockHeight)!
             )
 
-            emit ExecutionNodeVersionTableDeletion(height: blockHeight, version: removed)
-
+            //emit NodeVersionTableDeletion(height: blockHeight, version: removed)
             // Clean up upcoming & archived block boundaries before
             // returning removed version for verification
-            ExecutionNodeVersionBeacon.archiveOldBlockBoundaries()
+            NodeVersionBeacon.archiveOldBlockBoundaries()
             return removed
         }
 
+        /// Separate method to emit the NodeVersionTableUpdated event
+        ///
+        /// @param newVersions
+        /// @param deletedVersions
+        ///
+        pub fun emitNodeVersionTableUpdated() {
+            
+            let updatedTable: {UInt64: Semver} = {}
+
+            // Assure that there is at least one working version for the current block
+            let currentVersionBlock = NodeVersionBeacon.getNextVersionBoundaryPair()[0] as! UInt64
```

Comment:
Why index 0? Could this be a constant somewhere?

Reply:
I needed to look for the block for the current Version on the table, and Gio already had done that method that returns a 2 position array, where the first position is that block. I was trying to introduce minimal changes in this contract but I can agree that this does not look pretty at all